src/test/java/com/openlattice/rehearsal/ThrowingCallAdapterFactory.kt

```diff
@@ -25,41 +25,57 @@ import com.google.common.base.Charsets
 import retrofit2.Call
 import retrofit2.CallAdapter
 import retrofit2.Retrofit
+import java.lang.reflect.UndeclaredThrowableException
 
-class ThrowingCallAdapterFactory: CallAdapter.Factory() {
+class ThrowingCallAdapterFactory : CallAdapter.Factory() {
     companion object {
-        private val logger = LoggerFactory.getLogger( ThrowingCallAdapterFactory::class.java )
+        private val logger = LoggerFactory.getLogger(ThrowingCallAdapterFactory::class.java)
     }
 
-    override fun get(returnType: Type, annotations: Array<kotlin.Annotation>, retrofit: Retrofit): CallAdapter<*, *> =
-        object : CallAdapter<Any,Any>  {
-            override fun  responseType():Type {
-                return returnType
-            }
+    override fun get(returnType: Type, annotations: Array<Annotation>, retrofit: Retrofit): CallAdapter<*, *> =
+            object : CallAdapter<Any, Any> {
+                override fun responseType(): Type {
+                    return returnType
+                }
 
-            override fun adapt( call:Call<Any> ):Any? {
-                try {
-                    val response = call.execute()
-                    if ( response.code() >= 400 ) {
-                        val errorBody = IOUtils.toString( response.errorBody()!!.byteStream(), Charsets.UTF_8 )
-                        logger.error( "Call failed with code {} and message {} and error body {}",
-                                response.code(),
-                                response.message(),
-                                errorBody )
-                        throw GeneralException( errorBody )
+                override fun adapt(call: Call<Any>): Any? {
+                    return try {
+                        val response = call.execute()
+                        if (response.code() >= 400) {
+                            val errorBody = IOUtils.toString(response.errorBody()!!.byteStream(), Charsets.UTF_8)
+                            logger.error("Call failed with code {} and message {} and error body {}",
+                                    response.code(),
+                                    response.message(),
+                                    errorBody)
+                            throw GeneralException(errorBody)
+                        }
+                        response.body()
+                    } catch (e: IOException) {
+                        logger.error("Call failed.", e)
+                        null
                     }
-                    return response.body()
-                } catch ( e:IOException ) {
-                    logger.error( "Call failed.", e )
-                    return null
                 }
-            }
 
-        }
+            }
 }
 
-class GeneralException(message: String):java.lang.Exception(message)
\ No newline at end of file
+class GeneralException(message: String) : java.lang.Exception(message)
+
+internal fun assertException(fqn: () -> (Any), expectedMsg: String) {
```

Comment:
Maybe consider making this an `inline` function. I'm not sure if it's too big of a function for it but it should offset any performance penalty from passing that `()->(Any)` lambda in and it's called quite a lot

Reply:
👍 Cool, will do!